sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/simpleCosting.scala

```diff
@@ -36,22 +36,26 @@ case class SimpleCost(value: Long) extends Cost {
 }
 
 /**
- * A skew join aware implementation of [[CostEvaluator]], which counts the number of
- * [[ShuffleExchangeLike]] nodes and skew join nodes in the plan.
+ * A skew aware implementation of [[CostEvaluator]], which counts the number of
+ * skew join nodes and skew [[AQEShuffleReadExec]] nodes in the plan.
  */
-case class SimpleCostEvaluator(forceOptimizeSkewedJoin: Boolean) extends CostEvaluator {
+case class SimpleCostEvaluator(
+    forceOptimizeSkewedJoin: Boolean,
+    forceOptimizeSkewedShuffleRead: Boolean)
+    extends CostEvaluator {
   override def evaluateCost(plan: SparkPlan): Cost = {
     val numShuffles = plan.collect {
       case s: ShuffleExchangeLike => s
     }.size
 
-    if (forceOptimizeSkewedJoin) {
-      val numSkewJoins = plan.collect {
-        case j: ShuffledJoin if j.isSkewJoin => j
+    if (forceOptimizeSkewedJoin || forceOptimizeSkewedShuffleRead) {
+      val numSkews = plan.collect {
+        case j: ShuffledJoin if forceOptimizeSkewedJoin && j.isSkewJoin => j
+        case r: AQEShuffleReadExec if forceOptimizeSkewedShuffleRead && r.hasSkewedPartition => r
```

Comment:
if ShuffledJoin is skewed, it should always have AQEShuffleReadExec with skewed partition, right ? Then we will go into this pattern even if people disable force optimize skewed join..

Reply:
Makes sense, I will add relevant test cases

Reply:
I made the fix, but couldn't think of a suitable test case to cover it.